client/lib/signup/progress-store.js

```diff
@@ -4,139 +4,85 @@
  * External dependencies
  */
 
-import { assign, clone, filter, find, get, isEmpty, map, omit } from 'lodash';
+import { cloneDeep, isEqual, get, isEmpty, omit } from 'lodash';
 import debugFactory from 'debug';
 const debug = debugFactory( 'calypso:signup-progress-store' );
-import store from 'store';
 
 /**
  * Internal dependencies
  */
 import Dispatcher from 'dispatcher';
-import emitter from 'lib/mixins/emitter';
-import SignupDependencyStore from './dependency-store';
-import flows from 'signup/config/flows';
 import steps from 'signup/config/steps-pure';
-
-/**
- * Constants
- */
-const STORAGE_KEY = 'signupProgress';
-
-/**
- * Module variables
- */
-let signupProgress = [];
+import { SIGNUP_COMPLETE_RESET, SIGNUP_PROGRESS_SET } from 'state/action-types';
+import {
+	completeStep,
+	processStep,
+	removeUnneededSteps,
+	saveStep,
+	invalidateStep,
+	submitStep,
+} from 'state/signup/progress/actions';
+import { getSignupProgress } from 'state/signup/progress/selectors';
+import SignupDependencyStore from './dependency-store';
 
 const SignupProgressStore = {
-	get: function() {
-		return clone( signupProgress );
+	currentValue: null,
+	reduxStore: null,
+	subscribers: new Map(),
+
+	get() {
+		if ( ! this.reduxStore ) {
+			throw new Error( 'Tried to read redux state before store was ready' );
+		}
+		return cloneDeep( getSignupProgress( this.reduxStore.getState() ) );
 	},
-	reset: function() {
-		signupProgress = [];
-		store.remove( STORAGE_KEY );
+	reset() {
+		this.unsubscribeAll();
+		this.reduxStore.dispatch( {
+			type: SIGNUP_COMPLETE_RESET,
+		} );
 	},
-	getFromCache: function() {
-		loadProgressFromCache();
-		return this.get();
+	set( input ) {
+		this.reduxStore.dispatch( {
+			type: SIGNUP_PROGRESS_SET,
+			steps: input,
+		} );
+	},
+	setReduxStore( reduxStore ) {
```

Comment:
I think you can use the `redux-bridge` instead of having to hold onto a ref to the redux store.

Reply:
It seems like we'd want to continue using the `setReduxStore` pattern in the signup (for now) to preserve consistency with other signup stores. 

Also, it seems like value change subscription pattern is not yet supported in `redux-bridge`, so it might not be sufficient here without additional work (e.g. `SignupProgressStore.on( 'change', callbackFn );`). I might be wrong, though.